python_modules/dagster/dagster_tests/core_tests/execution_tests/test_run_metrics_thread.py

```diff
@@ -47,17 +49,26 @@ def test_get_container_metrics(mock_containerized_utilization_metrics):
         "dagster._core.execution.run_metrics_thread.retrieve_containerized_utilization_metrics",
         return_value=mock_containerized_utilization_metrics,
     ):
-        metrics = run_metrics_thread._get_container_metrics()  # noqa: SLF001
+        metrics = run_metrics_thread._get_container_metrics(  # noqa: SLF001
+            previous_cpu_usage_ms=49200, previous_measurement_timestamp=1000
+        )
 
         assert metrics
         assert isinstance(metrics, dict)
-        assert metrics["container.cpu_usage_ms"] == 50
+        assert metrics["container.cpu_usage_ms"] == 50000
         assert metrics["container.cpu_limit_ms"] == 100
         assert metrics["container.cpu_cfs_period_us"] == 10000
         assert metrics["container.cpu_cfs_quota_us"] == 1000
         assert metrics["container.memory_usage"] == 16384
         assert metrics["container.memory_limit"] == 65536
         assert metrics["container.memory_percent"] == 25
+        assert metrics["measurement_timestamp"] == 1010
+
+        # Calculation of expected cpu_percent
+        # cpu_limit_ms = 100 ms per second-> derived from cpu cfs quota and period
+        # cpu_usage_rate_ms = (50000 - 49200) / (1010 - 1000) = 80 ms per second
+        # cpu_percent = 100% * 80 (used) / 100 (limit) = 80.0%
+        assert metrics["container.cpu_percent"] == 80.0
```

Comment:
after testing with longer runs (constraining cpu to ridiculous amount), i realized i had two things wrong in my cpu usage and percent calculation. last commit fix that and adds test